### ISSUE TITLE
mAP defined in evaluation not showed in logs

### ISSUE THREAD

**Issue by FrancescoSaverioZuppichini** [open-mmlab/mmdetection#6175]
Thanks for your error report and we appreciate it a lot.

**Describe the bug**
Train a model with

```
evaluation = dict(interval=1, metric="mAP")
workflow = [("train", 1), ("val", 1)]

```

The logs don't show the mAP

```json
{"mode": "val", "epoch": 1, "iter": 48, "lr": 0.01, "memory": 5774, "data_time": 0.05716, "loss_cls": 0.53878, "loss_bbox": 0.99966, "loss_centerness": 0.62669, "loss": 2.16513, "time": 0.36834}
```

**Environment**

```{'sys.platform': 'linux', 'Python': '3.9.5 (default, Jun  4 2021, 12:28:51) [GCC 7.5.0]', 'CUDA available': True, 'GPU 0,1,2': 'GeForce GTX 1080 Ti', 'CUDA_HOME': '/usr/local/cuda', 'NVCC': 'Build cuda_11.2.r11.2/compiler.29373293_0', 'GCC': 'gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0', 'PyTorch': '1.9.1+cu102', 'PyTorch compiling details': 'PyTorch built with:\n  - GCC 7.3\n  - C++ Version: 201402\n  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications\n  - Intel(R) MKL-DNN v2.1.2 (Git Hash 98be7e8afa711dc9b66c8ff3504129cb82013cdb)\n  - OpenMP 201511 (a.k.a. OpenMP 4.5)\n  - NNPACK is enabled\n  - CPU capability usage: AVX2\n  - CUDA Runtime 10.2\n  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70\n  - CuDNN 7.6.5\n  - Magma 2.5.2\n  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=10.2, CUDNN_VERSION=7.6.5, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.9.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, \n', 'TorchVision': '0.10.1+cu102', 'OpenCV': '4.5.3', 'MMCV': '1.3.13', 'MMCV Compiler': 'GCC 9.3', 'MMCV CUDA Compiler': '11.2'}
```

**Comment by PeterVennerstrom** [comment]
The logs show losses during the training phase of an epoch and mAP is calculated on the validation set at some interval, e.g. after every epoch.

**Comment by PeterVennerstrom** [comment]
[Here are the docs](https://github.com/open-mmlab/mmdetection/blob/a616886bf1e8de325e6906b8c76b6a4924ef5520/docs/tutorials/customize_runtime.md) for runtime hooks.  

```
evaluation = dict(interval=1, metric="mAP")
```

The eval hook should be called every epoch by default unless the --no-validate argument is passed to train.py when starting training and interval=1 and metric="mAP" should be default settings in the code so neither need to be set by config unless a different behavior is needed.   

'''
workflow = [("train", 1), ("val", 1)]
'''

From the docs it looks like the "val" setting is an after validation hook that only logs the losses on the validation set by default.  It is different than the evaluation hook which is calculates mAP on the validation set.

```
workflow = [('train', 1)]
```

This line should be the only config setting needed to log accuracy metrics (specific to the dataset used) for the validation set after every training epoch.

**Comment by hhaAndroid** [comment]
> [Here are the docs](https://github.com/open-mmlab/mmdetection/blob/a616886bf1e8de325e6906b8c76b6a4924ef5520/docs/tutorials/customize_runtime.md) for runtime hooks.
> 
> ```
> evaluation = dict(interval=1, metric="mAP")
> ```
> 
> The eval hook should be called every epoch by default unless the --no-validate argument is passed to train.py when starting training and interval=1 and metric="mAP" should be default settings in the code so neither need to be set by config unless a different behavior is needed.
> 
> ```
> workflow = [("train", 1), ("val", 1)]
> ```
> 
> From the docs it looks like the "val" setting is an after validation hook that only logs the losses on the validation set by default. It is different than the evaluation hook which calculates mAP on the validation set.
> 
> ```
> workflow = [('train', 1)]
> ```
> 
> This line should be the only config setting needed to log accuracy metrics (specific to the dataset used) for the validation set after every training epoch.

Yes
